### PR TITLE
junest: init at 7.4.8

### PR DESCRIPTION
**Pull request by jdev082**
## Description of changes
A program called [junest](https://github.com/fsquillace/junest) that was requested to be packaged in https://github.com/NixOS/nixpkgs/issues/304461.  "The lightweight Arch Linux based distro that runs, without root privileges, on top of any other Linux distro."

## Things done
**Note: Since this program is written in bash, it is likely to be architecture agnostic. Architectures selected here are of the system(s) that this package have been tested with.**

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by vinnymeller — CHANGES_REQUESTED** [review]
### Commit messages

Make sure your commits follow the conventions for nixpkgs, https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions

In this case, you'll want your commits squashed to 1 with the same 1st line as your pr title, `junest: init at 7.4.8`

### nixpkgs-review

Just because PRs often take a while to get to, I recommend running [nixpkgs-review](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#tested-compilation-of-all-pkgs-that-depend-on-this-change-using-nixpkgs-review) yourself - other people can always verify themselves later as they please.

In this case, running it gives me this output:

Result of `nixpkgs-review pr 305418` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

This indicates that this PR isn't actually adding the package as expected


### Directory structure

With this derivation defined where it is, you'd need an entry in [this file](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix#L9477) to expose the package.

_But_, I'm sure someone will correct me if I'm wrong, my understanding is that _most_ new packages should go in `pkgs/by-name/`, so this derivation would go in `pkgs/by-name/ju/junest/package.nix`

With the file there, you don't need to add an entry to a `top-level` file, it'll get picked up automagically

This new(er) structure comes from [RFC 140](https://github.com/NixOS/rfcs/blob/master/rfcs/0140-simple-package-paths.md)


###  Building and testing

Once your package is exposed, from your local nixpkgs, you can run `nix build .#junest`, which will create a `results` directory. In this case there should be `out/bin/junest` and the other dirs that exist

You / we'll be wanting to test that output to ensure everything works as expected.

You will find that currently there are some issues with how things are structured when you try to build/test

File: pkgs/os-specific/linux/junest/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{ lib, stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  version = "7.4.8";
+  pname = "junest";
+  src = fetchFromGitHub {
+    owner = "fsquillace";
+    repo = "junest";
+    rev = "refs/tags/${version}";
+    sha256 = "sha256-9yrQ721fHUxXEZ0mh27SB8yoUH67ltOktUq3kr4qrBc=";
```
Comment (kirillrdy) [inline pkgs/os-specific/linux/junest/default.nix:10]:
```suggestion
    hash = "sha256-9yrQ721fHUxXEZ0mh27SB8yoUH67ltOktUq3kr4qrBc=";
```

https://nixos.org/manual/nixpkgs/unstable/#fetchfromgithub

**Comment by jdev082** [comment]
Something went wrong. Creating a new PR.

**Comment by jdev082** [comment]
Just fixing some stuff...

**Comment by jdev082** [comment]
Something went wrong. Creating a new PR.

**Comment by jdev082** [comment]
https://github.com/NixOS/nixpkgs/pull/306644

**Pull request closed without merging**